src/Dependency.cpp

```diff
@@ -107,72 +63,85 @@ Dependency::Dependency(std::string path)
     }
 
     // check if given path is a symlink
-    if (original_file != rtrim(path))
```

Comment:
General note, I wish this had been submitted as a series of smaller pull requests. This pull request is so large it basically rewrites the entire app into a different product, that is exceedinly hard to review for me. I'm asking this without any sarcasm, if I merge this PR there's basically very little of my old code left and this would become your app basically, so would you actually be willing to take ownership of it and maintain it? I wrote this as a small utility years ago and no longer have time/interest to maintain it so TBH if you're interested in doing huge changes it might actually be simple to actually make you maintainer of this app

Reply:
Sure, I'm willing to maintain if you'd like. I understand it's a relatively large PR, so how about creating a new branch for it while I work on breaking this pull request into smaller PRs?

Reply:
This way it can be more thoroughly tested and gain additional feedback/review.

Reply:
I've invited you as a collaborator, as I'm more than happy to have help on this. But yes, I like your idea of creating a branch and breaking it into smaller PRs. After all, it is good practice to keep commits fairly focused around one thing - makes things much easier to follow than huge commits that contain unrelated changes

Reply:
Thank you for the invitation. I'm more than happy to help.

Reply:
Ah, another suggestion : the Qt stuff seems quite mixed/intertwined with other code. It might be preferable to move Qt stuff to its own file. Mixing Qt-specific code within the generic bundling risks putting too much unrelated stuff in the same file

Reply:
Completely agree.

Reply:
@SCG82 We're using your fork on OBS for GitHub Actions CI right now and it works without issues, but when trying to bundle the app with self-compiled Qt Frameworks I run into errors. You have issues disabled on your fork, what would be the best way to reach out/report?

Reply:
I enabled issues on my fork. Thanks.